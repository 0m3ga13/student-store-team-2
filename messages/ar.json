{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
