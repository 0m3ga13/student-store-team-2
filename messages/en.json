{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
