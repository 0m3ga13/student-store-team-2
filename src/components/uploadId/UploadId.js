"use client";
import {
    getDownloadURL,
    getStorage,
    ref,
    uploadBytesResumable,
} from "firebase/storage";
import Image from "next/image";
import { useTranslations } from "next-intl";
import { useState } from "react";
import { BsFillPersonVcardFill } from "react-icons/bs";

import { useUser } from "../userProvider/UserProvider";

const UploadId = ({ onIdUpload, profile_id }) => {
    const t = useTranslations("Index");
    const [idFile, setIdFile] = useState(null);
    const [downloadURL, setDownloadURL] = useState(profile_id);
    const [isUploading, setIsUploading] = useState(false);
    const [progressUpload, setProgressUpload] = useState(0);
    const user = useUser();
    const storage = getStorage();

    const handleSelectedFile = (files) => {
        if (files && files[0].size < 10000000) {
            setIdFile(files[0]);
        } else {
            console.error("The file size is too large or no file selected.");
        }
    };

    const handleUploadFile = () => {
        if (idFile) {
            const storageRef = ref(storage, `ID/${user.user}`);
            const uploadTask = uploadBytesResumable(storageRef, idFile);

            setIsUploading(true); // Start the spinner

            uploadTask.on(
                "state_changed",
                (snapshot) => {
                    const progress =
                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
                    setProgressUpload(progress);

                    switch (snapshot.state) {
                        case "paused":
                            console.log("Upload is paused");
                            break;
                        case "running":
                            console.log("Upload is running");
                            break;
                    }
                },
                (error) => {
                    console.error(error.message);
                    setIsUploading(false); // Stop the spinner if an error occurs
                },
                () => {
                    getDownloadURL(uploadTask.snapshot.ref)
                        .then((url) => {
                            setDownloadURL(url);
                            setIsUploading(false);
                            onIdUpload(url); // Send the URL to the parent component
                        })
                        .catch((error) => {
                            console.error(
                                "Error getting download URL: ",
                                error
                            );
                            setIsUploading(false);
                        });
                }
            );
        } else {
            console.error("File not found");
        }
    };

    const handleRemoveFile = () => setIdFile(null);

    return (
        <div className='flex justify-start ml-10 mb-4 items-center space-x-5'>
            {downloadURL ? (
                <Image
                    width={50}
                    height={50}
                    src={downloadURL}
                    alt='id'
                    className='w-24 h-24 rounded-md '
                />
            ) : (
                <>
                    <BsFillPersonVcardFill className='w-24 h-24  text-red-500 rounded-md' />
                </>
            )}
            <div>
                <label
                    htmlFor='upload-button-id'
                    className='bg-none border border-accent rounded-md text-accent px-5 py-2 tracking-tight font-bold shadow-sm cursor-pointer'
                >
                    {downloadURL ? t("Edit ID") : t("Upload ID")}
                </label>
                <input
                    type='file'
                    id='upload-button-id'
                    style={{ display: "none" }}
                    onChange={(e) => handleSelectedFile(e.target.files)}
                />
                {isUploading ? (
                    <div className='text-primary' role='status'>
                        <span className='visually-hidden'>
                            {t("Loading")}...
                        </span>
                    </div>
                ) : (
                    idFile && (
                        <div className='m-2'>
                            <p>{idFile.name}</p>
                            <p>{`Size: ${(idFile.size / (1024 * 1024)).toFixed(
                                2
                            )} MB`}</p>
                            <button
                                className='bg-red-500 text-white rounded-md px-3 py-1'
                                onClick={handleRemoveFile}
                            >
                                {t("Close")}
                            </button>
                            <button
                                className='border border-blue-500 ml-2 text-blue-500 rounded-md px-3 py-1'
                                onClick={handleUploadFile}
                            >
                                {t("Upload")}
                            </button>
                        </div>
                    )
                )}
            </div>
        </div>
    );
};

export default UploadId;
