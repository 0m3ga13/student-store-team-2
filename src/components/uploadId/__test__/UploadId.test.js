import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import UploadId from "../UploadId";
import messages from "../../../../messages/en.json";
jest.mock("firebase/storage", () => {
    return {
        getDownloadURL: jest.fn(),
        getStorage: jest.fn(),
        ref: jest.fn(),
        uploadBytesResumable: jest.fn(),
    };
});
jest.mock("firebase/auth", () => {
    const authInstance = {
        // Add mock methods and properties as needed
    };

    return {
        getAuth: jest.fn(() => authInstance),
        createUserWithEmailAndPassword: jest.fn(),
        signInWithEmailAndPassword: jest.fn(),
        // Add other authentication methods used in your code
        GoogleAuthProvider: jest.fn(),
    };
});
jest.mock("firebase/firestore", () => {
    return {
        getFirestore: jest.fn(),
        doc: jest.fn(),
        setDoc: jest.fn(),
        // Add other Firestore methods used in your code
    };
});
jest.mock("@supabase/supabase-js", () => {
    return {
        ...jest.requireActual("@supabase/supabase-js"), // Use the actual module for non-mocked parts
        createClient: jest.fn(() => ({
            from: jest.fn(() => ({
                select: jest.fn(() => ({
                    eq: jest.fn(() => ({
                        // Customize the behavior of your Supabase queries as needed
                        data: { id: 123, name: "John Doe" },
                        error: null,
                    })),
                })),
            })),
        })),
    };
});
it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <UploadId />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
