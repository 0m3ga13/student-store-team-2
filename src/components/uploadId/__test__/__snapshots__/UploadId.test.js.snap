// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-start ml-10 mb-4 items-center space-x-5"
>
  <svg
    className="w-24 h-24  text-red-500 rounded-md"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V4Zm9 1.5a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 0-1h-4a.5.5 0 0 0-.5.5ZM9 8a.5.5 0 0 0 .5.5h4a.5.5 0 0 0 0-1h-4A.5.5 0 0 0 9 8Zm1 2.5a.5.5 0 0 0 .5.5h3a.5.5 0 0 0 0-1h-3a.5.5 0 0 0-.5.5Zm-1 2C9 10.567 7.21 9 5 9c-2.086 0-3.8 1.398-3.984 3.181A1 1 0 0 0 2 13h6.96c.026-.163.04-.33.04-.5ZM7 6a2 2 0 1 0-4 0 2 2 0 0 0 4 0Z"
    />
  </svg>
  <div>
    <label
      className="bg-none border border-accent rounded-md text-accent px-5 py-2 tracking-tight font-bold shadow-sm cursor-pointer"
      htmlFor="upload-button-id"
    >
      Upload ID
    </label>
    <input
      id="upload-button-id"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
</div>
`;
